test/common/runtime/runtime_impl_test.cc

```diff
@@ -203,6 +203,45 @@ TEST_F(DiskBackedLoaderImplTest, OverrideFolderDoesNotExist) {
   EXPECT_EQ("hello", loader->snapshot().get("file1"));
 }
 
+TEST_F(DiskBackedLoaderImplTest, PercentHandling) {
+  setup();
+  run("test/common/runtime/test_data/current", "envoy_override");
+
+  envoy::type::FractionalPercent default_value;
+
+  // Smoke test integer value of 0, should be interpreted as 0%
+  {
+    loader->mergeValues({{"foo", "0"}});
+
+    EXPECT_FALSE(loader->snapshot().featureEnabled("foo", default_value, 0));
+    EXPECT_FALSE(loader->snapshot().featureEnabled("foo", default_value, 5));
+  }
+
+  // Smoke test integer value of 5, should be interpreted as 5%
+  {
+    loader->mergeValues({{"foo", "5"}});
+    EXPECT_TRUE(loader->snapshot().featureEnabled("foo", default_value, 0));
+    EXPECT_TRUE(loader->snapshot().featureEnabled("foo", default_value, 4));
+    EXPECT_FALSE(loader->snapshot().featureEnabled("foo", default_value, 5));
+    EXPECT_TRUE(loader->snapshot().featureEnabled("foo", default_value, 100));
+  }
+
+  // Verify uint64 -> uint32 conversion by using a runtime value with all 0s in
+  // the bottom 32 bits. If it were to be naively treated as a uint32_t then it
+  // would appear as 0%, but it should be 100% because we assume the
+  // denominator is 100
+  {
+    uint64_t high_value = 1UL << 60;
```

Comment:
nit: What you have works, but you could also just set it to:
`std::numeric_limits<uint64_t>::max()`

Reply:
it needs to have the property that `low_32_bits(high_value) % 100 < 100`  which is why i did it that way, 60 was arbitrary though it just needed to be bigger than a uint32

Reply:
i'll put this in a comment